### PR TITLE
PLAT-601-7 Implemented User Localization Presets

### PR DESCRIPTION
**Pull request by oliver-richers**
image
image
image
image
image
image

### CODE REVIEW

File: platform-core-module/src/main/java/com/softicar/platform/core/module/CoreI18n.java [hunk 1]
```diff
@@ -361,6 +363,12 @@
 		.de("Lokaler Name");
 	I18n0 LOCAL_PORT = new I18n0("Local Port")//
 		.de("Lokaler Port");
+	I18n0 LOCALIZATION = new I18n0("Localization")//
+		.de("Lokalisation");
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/CoreI18n.java:367]: I think this should be `Lokalisierung` (much like above and below)

File: platform-common-core/src/main/java/com/softicar/platform/common/core/CommonCoreI18n.java [hunk 2]
```diff
@@ -5,6 +5,10 @@
 
 public interface CommonCoreI18n {
 
+	I18n0 DECIMAL_SEPARATOR = new I18n0("Decimal Separator")//
+		.de("Dezimaltrennzeichen");
+	I18n0 DIGIT_GROUP_SEPARATOR = new I18n0("Digit Group Separator")//
+		.de("Zifferngruppenseparator");
```
Comment (alexander-schmidt) [inline platform-common-core/src/main/java/com/softicar/platform/common/core/CommonCoreI18n.java:11]: I think this should be `Zifferngruppentrennzeichen`, for the sake of symmetry

File: platform-core-module/src/main/java/com/softicar/platform/core/module/CoreI18n.java [hunk 3]
```diff
@@ -985,6 +993,6 @@
 		.de("Ihr aktuelles Passwort ist zu alt.");
 	I18n0 YOUR_CURRENT_PASSWORD_MIGHT_HAVE_BEEN_COMPROMISED = new I18n0("Your current password might have been compromised.")//
 		.de("Ihr aktuelles Passwort wurde möglicherweise kompromittiert.");
-	I18n1 YOUR_LANGUAGE_HAS_BEEN_CHANGED_TO_ARG1 = new I18n1("Your language has been changed to '%s'.")//
-		.de("Ihre Sprache wurde geändert auf '%s'.");
+	I18n0 YOUR_LOCALIZATION_HAS_BEEN_CHANGED = new I18n0("Your localization has been changed.")//
+		.de("Ihre Lokalisation wurde geändert.");
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/CoreI18n.java:997]: see above; we should settle for either `Lokalisation` or `Lokalisierung`

File: platform-core-module/src/main/java/com/softicar/platform/core/module/user/profile/UserProfileChangeLocalizationDiv.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+		public void handleClick() {
+
+			attributesDiv.tryToApplyValidateAndSave();
+
+			CurrentLocale.set(CurrentUser.get().getLocale());
+
+			executeAlert(
+				CoreI18n.YOUR_LOCALIZATION_HAS_BEEN_CHANGED//
+					.concatSentence(CoreI18n.PLEASE_PRESS_F5_TO_REFRESH));
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/user/profile/UserProfileChangeLocalizationDiv.java:43]: (Beyond this PR: Pressing F5 should not be necessary - that's poor UX. We could automate this quite easily.)

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-core-module/src/main/java/com/softicar/platform/core/module/locale/AGLocalizationPreset.sqml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+PACKAGE com.softicar.platform.core.module.locale
+
+IMPORT com.softicar.platform.core.module.language.AGCoreLanguage
+
+TABLE AGLocalizationPreset 'Core.LocalizationPreset' {
+	PK Integer id = SERIAL
+	String name [MAXLENGTH=255]
+	
+	AGCoreLanguage language
+	String decimalSeparator = '.' [MAXLENGTH=255]
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/locale/AGLocalizationPreset.sqml:10]: I don't think we should define a default here.

File: platform-core-module/src/main/java/com/softicar/platform/core/module/locale/AGLocalizationPreset.sqml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+	String digitGroupSeparator = '' [MAXLENGTH=255]
```
Comment (alexander-schmidt) [inline platform-core-module/src/main/java/com/softicar/platform/core/module/locale/AGLocalizationPreset.sqml:11]: I don't think we should define a default here.

File: platform-core-module/src/main/java/com/softicar/platform/core/module/CoreI18n.java (hunk 1 above)
Comment (oliver-richers) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/CoreI18n.java:367]: done

File: platform-core-module/src/main/java/com/softicar/platform/core/module/CoreI18n.java (hunk 3 above)
Comment (oliver-richers) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/CoreI18n.java:997]: done

File: platform-core-module/src/main/java/com/softicar/platform/core/module/user/profile/UserProfileChangeLocalizationDiv.java (hunk 4 above)
Comment (oliver-richers) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/user/profile/UserProfileChangeLocalizationDiv.java:43]: true, but please in a separate story

File: platform-core-module/src/main/java/com/softicar/platform/core/module/locale/AGLocalizationPreset.sqml (hunk 5 above)
Comment (oliver-richers) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/locale/AGLocalizationPreset.sqml:10]: well, done

File: platform-core-module/src/main/java/com/softicar/platform/core/module/locale/AGLocalizationPreset.sqml (hunk 6 above)
Comment (oliver-richers) [reply platform-core-module/src/main/java/com/softicar/platform/core/module/locale/AGLocalizationPreset.sqml:11]: redundant ;)

File: platform-common-core/src/main/java/com/softicar/platform/common/core/CommonCoreI18n.java (hunk 2 above)
Comment (oliver-richers) [reply platform-common-core/src/main/java/com/softicar/platform/common/core/CommonCoreI18n.java:11]: absolutely right, thx

**Review by alexander-schmidt — APPROVED** [review]

**Pull request merged**